docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,215 @@
+
+## Use-Cases
+
+### Backup Operation
+- As a non-admin user/namespace owner with administrative privileges for a particular namespace, the user should be able to:
+    - Create a Backup of the namespace
+    - View the status of the Backup created for the particular namespace
+    - Delete the Backup of the namespace
+
+### Restore Operation
+- As a non-admin user/namespace owner with administrative privileges for a particular namespace, the user should be able to:
+    - Create a Restore of the namespace
+    - View the status of the Restore created for the particular namespace
+    - Delete the Restore of the namespace
+
+### BackupStorageLocation (BSL) Configuration Operation (optional use case)
+- As a non-admin user/namespace owner with administrative privileges for a particular namespace, the user should be able to:
+    - Create their own BSLs
+    - View the status of the BSLs created by them
+    - Delete the BSLs created by them
+
+## Installation
+- The Non-Admin Controller (NAC) will be installed via OADP Operator.
+- The Data Protection Application (DPA) CR will consist of a root level struct called `nonAdmin` under the features. This struct will have a boolean to `enable` or `disable` the non-admin feature.
+- If the `features.nonAdmin.enable` flag is set to `true`, the OADP Operator will install the NAC in OADP Operator's install namespace, by default this feature will be disabled.
+- **Multiple OADP:NAC installations:** Due to performance limitations of single threaded nature of Velero controller, it might be desirable in some environments to have more than one instance of OADP:NAC installed.
+  - We need to ensure that one OADP operator instance is mapped to only one NAC instance
+  - Also, the namespaces catered by each NAC instance should be separate and not merge with each other
+  - We can facilitate this by accepting a list of NS that will be watched/catered by the particular NAC instance. We can do this during the installation time via DPA. There may be 2 ways which we will support of NS filtering:
+    - Direct List of NS Names `(Spec.features.nonAdmin.NSList[])`
+    - List of NS to filtered based on a LabelSelector `(Spec.features.nonAdmin.labelSelector)`
+    - By default, of none of the above are specified then NAC listens to all the Namespaces in the cluster
+    - We need to error out if NAC NS scopes overlap (i.e. NS watch list overlap)
+
+
+## Pre-requisites
+- **OADP installed**: OADP Operator must be installed and configured to use non-admin controller
+- **Non Admin Controller configured**: Data Protection Application (DPA) instance must configure Non Admin Controller to watch user namespace(s), by default it watches all Namespaces
+- **RBAC privileges for the user**: User must have the appropriate RBAC privileges to create Non Admin objects within the Namespace where Backup will be taken. An example of such ClusterRole, which may be added to the user with `RoleBinding`:
+//TODO: This should be a Role ?
```

Comment:
I think does not need to be a Role instead of ClusterRole

For example, the way I am creating adding RBAC to non admin users I currently create is with this command
```sh
oc create rolebinding <non-admin-username>-namespace-admin --clusterrole=admin --user=<non-admin-username> --namespace=<non-admin-user-namespace>
```
this way, I am adding permission only to the namespace I specify

Reply:
Yeah it was an old todo